### ISSUE TITLE
GitHub pages returning content-type application/octet-stream for .html

### ISSUE THREAD

**Issue by xvaara** [antfu/vite-ssg#331]
### Describe the bug

vite-ssg creates a ".html" names file for the route /
using github pages it is served as index using wrong content-type

```
# curl -i https://xvaara.github.io/vite-ssg-ghpages/
HTTP/2 200
server: GitHub.com
content-type: application/octet-stream
```

github support suggested that it's not a bug on their end, so opening a ticket.

seems that vite-plugin-vue-layouts causes some url to be generated. without it the .html file isn't created.

### Reproduction

https://github.com/xvaara/vite-ssg-ghpages

### System Info

```Shell
not relavant
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by xvaara** [comment]
seems that there is a pull request to fix this in vite-plugin-vue-layouts:
https://github.com/JohnCampionJr/vite-plugin-vue-layouts/pull/97

still seems like a weird file to create :D

**Comment by YunYouJun** [comment]
@JohnCampionJr Let's get it going.

**Comment by xvaara** [comment]
fixed in vite-plugin-vue-layouts v0.8.0
